pkg/master/master.go

```diff
@@ -376,34 +361,49 @@ func logStackOnRecover(panicReason interface{}, httpWriter http.ResponseWriter)
 // init initializes master.
 func (m *Master) init(c *Config) {
 
-	nodeRESTStorage := minion.NewREST(m.minionRegistry)
+	boundPodFactory := &pod.BasicBoundPodFactory{}
+	podStorage, bindingStorage := podetcd.NewREST(c.EtcdHelper, boundPodFactory)
+	podRegistry := pod.NewRegistry(podStorage)
+
+	eventRegistry := event.NewEtcdRegistry(c.EtcdHelper, uint64(c.EventTTL.Seconds()))
+	limitRangeRegistry := limitrange.NewEtcdRegistry(c.EtcdHelper)
+	resourceQuotaRegistry := resourcequota.NewEtcdRegistry(c.EtcdHelper)
+	m.namespaceRegistry = namespace.NewEtcdRegistry(c.EtcdHelper)
+
+	// TODO: split me up into distinct storage registries
```

Comment:
It was splitting replication controller registry/rest/storage from service.  Advantage is they can be coupled independently.


> On Feb 13, 2015, at 6:12 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/master/master.go:
> 
> > @@ -376,34 +361,49 @@ func logStackOnRecover(panicReason interface{}, httpWriter http.ResponseWriter)
> >  // init initializes master.
> >  func (m *Master) init(c *Config) {
> >  
> > -	nodeRESTStorage := minion.NewREST(m.minionRegistry)
> > +	boundPodFactory := &pod.BasicBoundPodFactory{}
> > +	podStorage, bindingStorage := podetcd.NewREST(c.EtcdHelper, boundPodFactory)
> > +	podRegistry := pod.NewRegistry(podStorage)
> > +
> > +	eventRegistry := event.NewEtcdRegistry(c.EtcdHelper, uint64(c.EventTTL.Seconds()))
> > +	limitRangeRegistry := limitrange.NewEtcdRegistry(c.EtcdHelper)
> > +	resourceQuotaRegistry := resourcequota.NewEtcdRegistry(c.EtcdHelper)
> > +	m.namespaceRegistry = namespace.NewEtcdRegistry(c.EtcdHelper)
> > +
> > +	// TODO: split me up into distinct storage registries
> What disadvantage is there in using a single registry for everything?
> 
> —
> Reply to this email directly or view it on GitHub.
>